### PR TITLE
[ServiceBus] T2 Document Runtime Exception in public API

### LINKED ISSUE #11379: Update documentation to include runtime exceptions
* Update documentation to include any runtime exceptions
  * (ie. `send(Message)`)

### PR DESCRIPTION
**Pull request by hemanttanwar**
Document RuntimeExceptions in public API

fixes #11379

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java [hunk 1]
```diff
@@ -220,6 +223,8 @@ public String getEntityPath() {
      *     {@code transactionContext.transactionId} is null when {@code options.transactionContext} is specified.
      * @throws UnsupportedOperationException if the receiver was opened in
      *     {@link ServiceBusReceiveMode#RECEIVE_AND_DELETE} mode or if the message was received from peekMessage.
+     * @throws IllegalStateException if receiver is already disposed.
+     * @throws IllegalArgumentException if message is already been settled.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java:228]: Make the changes in other places too.

```suggestion
     * @throws IllegalArgumentException if the message is already settled.
```

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java [hunk 2]
```diff
@@ -808,6 +853,7 @@ public String getEntityPath() {
      *
      * @return The next expiration time for the session lock.
      * @throws IllegalStateException if the receiver is a non-session receiver.
+     * @throws IllegalStateException if receiver is already disposed.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java:863]: `@throws IllegalStateException` is duplicated. Combine them.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java [hunk 3]
```diff
@@ -185,6 +188,7 @@ public String getEntityPath() {
      * @throws NullPointerException if {@code batch}, {@code transactionContext} or
      * {@code transactionContext.transactionID} is {@code null}.
      * @throws AmqpException if {@code messages} is larger than the maximum allowed size of a single batch.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java:190]: Don't we wrap this AmqpException with ServiceBusException for this error?

Comment (hemanttanwar) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java:190]: yes we map it, let me change this to  SBE.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java [hunk 4]
```diff
@@ -1010,8 +1065,7 @@ private boolean isManagementToken(String lockToken) {
         } else if (message.getLockToken() == null) {
             // message must be a peeked message (or somehow they created a message w/o a lock token)
             final String errorMessage = "This operation is not supported for peeked messages. "
-                + "Only messages received using receiveMessages() or receiveMessagesWithContext() "
-                + "in PEEK_LOCK mode can be settled.";
+                + "Only messages received using receiveMessages() in PEEK_LOCK mode can be settled.";
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java:1068]: Can this string also be added to Message.java class along with all other error messages?

**Review by srnagar — APPROVED** [review]

**Pull request merged**